--- capicxx-someip-runtime.orig/CMakeLists.txt	2025-11-25 12:43:46.175394724 +0200
+++ capicxx-someip-runtime/CMakeLists.txt	2025-11-27 20:02:36.926608818 +0200
@@ -7,6 +7,22 @@
 
 project (CommonAPI-SomeIP)
 
+message(STATUS "Configuring for QNX...")
+
+# mmagdi - Ensure we are building for QNX
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "QNX")
+    message(FATAL_ERROR "This build supports only QNX.")
+endif()
+
+#Disable Linux syscalls in SOME/IP runtime
+# add_definitions(
+#     -DCAPI_NO_EPOLL
+#     -DCAPI_NO_EVENTFD
+#     -DCAPI_NO_TIMERFD
+#     -DCAPI_NO_INOTIFY
+# )
+##################################################
+
 set (CMAKE_VERBOSE_MAKEFILE off)
 
 set (LIBCOMMONAPI_SOMEIP_MAJOR_VERSION 3)
@@ -47,27 +63,32 @@
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-if (MSVC)
-# Boost
-find_package( Boost 1.55 COMPONENTS system thread log REQUIRED )
-if(Boost_FOUND)
-  if(Boost_LIBRARY_DIR)
-    MESSAGE( STATUS "Boost_LIBRARY_DIR not empty using it: ${Boost_LIBRARY_DIR}" )
-  else()
-    if(BOOST_LIBRARYDIR)
-      MESSAGE( STATUS "Boost_LIBRARY_DIR empty but BOOST_LIBRARYDIR is set setting Boost_LIBRARY_DIR to: ${BOOST_LIBRARYDIR}" )
-      set(Boost_LIBRARY_DIR ${BOOST_LIBRARYDIR})
-    endif()
-  endif()
-else()
-  MESSAGE( STATUS "Boost was not found!")
-endif()
-include_directories( ${Boost_INCLUDE_DIR} )
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /std:c++17 -D_CRT_SECURE_NO_WARNINGS /wd4503")
-link_directories(${Boost_LIBRARY_DIR})
-else()
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -Wall -Wextra -Wformat -Wformat-security -Wconversion -fexceptions -fstrict-aliasing -fstack-protector -fasynchronous-unwind-tables -fno-omit-frame-pointer -DCOMMONAPI_INTERNAL_COMPILATION -D_GLIBCXX_USE_NANOSLEEP -DBOOST_LOG_DYN_LINK -pthread -fvisibility=hidden")
-endif()
+# mmagdi - Commented out for QNX
+# if (MSVC)
+# # Boost
+# find_package( Boost 1.55 COMPONENTS system thread log REQUIRED )
+# if(Boost_FOUND)
+#   if(Boost_LIBRARY_DIR)
+#     MESSAGE( STATUS "Boost_LIBRARY_DIR not empty using it: ${Boost_LIBRARY_DIR}" )
+#   else()
+#     if(BOOST_LIBRARYDIR)
+#       MESSAGE( STATUS "Boost_LIBRARY_DIR empty but BOOST_LIBRARYDIR is set setting Boost_LIBRARY_DIR to: ${BOOST_LIBRARYDIR}" )
+#       set(Boost_LIBRARY_DIR ${BOOST_LIBRARYDIR})
+#     endif()
+#   endif()
+# else()
+#   MESSAGE( STATUS "Boost was not found!")
+# endif()
+# include_directories( ${Boost_INCLUDE_DIR} )
+# set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /std:c++17 -D_CRT_SECURE_NO_WARNINGS /wd4503")
+# link_directories(${Boost_LIBRARY_DIR})
+# else()
+# set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -Wall -Wextra -Wformat -Wformat-security -Wconversion -fexceptions -fstrict-aliasing -fstack-protector -fasynchronous-unwind-tables -fno-omit-frame-pointer -DCOMMONAPI_INTERNAL_COMPILATION -D_GLIBCXX_USE_NANOSLEEP -DBOOST_LOG_DYN_LINK -pthread -fvisibility=hidden")
+#endif()
+
+# mmagdi - Adjusted for QNX
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -Wall -fexceptions -pthread")
+##################################################
 
 SET(MAX_LOG_LEVEL "DEBUG" CACHE STRING "maximum log level")
 message(STATUS "MAX_LOG_LEVEL is set to value: ${MAX_LOG_LEVEL}")
@@ -179,6 +200,15 @@
 export (PACKAGE CommonAPI-SomeIP)
 
 # Create the CommonAPI-SomeIPConfig.cmake and CommonAPI-SomeIPConfigVersion files
+# mmagdi - Make sure absolute install paths exist before use
+if(NOT DEFINED ABSOLUTE_INSTALL_INCLUDE_DIR)
+    set(ABSOLUTE_INSTALL_INCLUDE_DIR "${CMAKE_INSTALL_PREFIX}/${INSTALL_INCLUDE_DIR}")
+endif()
+
+if(NOT DEFINED ABSOLUTE_INSTALL_CMAKE_DIR)
+    set(ABSOLUTE_INSTALL_CMAKE_DIR "${CMAKE_INSTALL_PREFIX}/${INSTALL_CMAKE_DIR}")
+endif()
+
 file (RELATIVE_PATH REL_INCLUDE_DIR "${ABSOLUTE_INSTALL_CMAKE_DIR}" "${ABSOLUTE_INSTALL_INCLUDE_DIR}")
 
 configure_file (${CMAKE_CURRENT_SOURCE_DIR}/cmake/CommonAPI-SomeIPConfig.cmake.in
@@ -202,14 +232,15 @@
 
 ##############################################################################
 # create pkg-config file
-if(NOT WIN32)
-        configure_file(CommonAPI-SomeIP.pc.in ${PROJECT_BINARY_DIR}/CommonAPI-SomeIP.pc @ONLY)
-        install(FILES ${PROJECT_BINARY_DIR}/CommonAPI-SomeIP.pc DESTINATION lib/pkgconfig)
-
-        # force all use of std::mutex and std::recursive_mutex to use runtime init
-        # instead of static initialization so mutexes can be hooked to enable PI as needed
-        add_definitions(-D_GTHREAD_USE_MUTEX_INIT_FUNC -D_GTHREAD_USE_RECURSIVE_MUTEX_INIT_FUNC)
-endif()
+# mmagdi - Commented out for QNX
+# if(NOT WIN32)
+#         configure_file(CommonAPI-SomeIP.pc.in ${PROJECT_BINARY_DIR}/CommonAPI-SomeIP.pc @ONLY)
+#         install(FILES ${PROJECT_BINARY_DIR}/CommonAPI-SomeIP.pc DESTINATION lib/pkgconfig)
+
+#         # force all use of std::mutex and std::recursive_mutex to use runtime init
+#         # instead of static initialization so mutexes can be hooked to enable PI as needed
+#         add_definitions(-D_GTHREAD_USE_MUTEX_INIT_FUNC -D_GTHREAD_USE_RECURSIVE_MUTEX_INIT_FUNC)
+# endif()
 
 ##############################################################################
 # maintainer-clean
