diff -ruN capicxx-someip-runtime.orig/include/CommonAPI/SomeIP/ClientId.hpp capicxx-someip-runtime/include/CommonAPI/SomeIP/ClientId.hpp
--- capicxx-someip-runtime.orig/include/CommonAPI/SomeIP/ClientId.hpp	2025-11-25 12:43:46.178650336 +0200
+++ capicxx-someip-runtime/include/CommonAPI/SomeIP/ClientId.hpp	2025-11-25 13:47:40.136478774 +0200
@@ -44,8 +44,8 @@
     COMMONAPI_EXPORT size_t hashCode();
 
     COMMONAPI_EXPORT client_id_t getClientId();
-    COMMONAPI_EXPORT uid_t getUid() const;
-    COMMONAPI_EXPORT gid_t getGid() const;
+    COMMONAPI_EXPORT CommonAPI::uid_t getUid() const override;
+    COMMONAPI_EXPORT CommonAPI::gid_t getGid() const override;
     COMMONAPI_EXPORT std::string getEnv() const;
     COMMONAPI_EXPORT std::string getHostAddress() const;
 
diff -ruN capicxx-someip-runtime.orig/include/CommonAPI/SomeIP/OutputStream.hpp capicxx-someip-runtime/include/CommonAPI/SomeIP/OutputStream.hpp
--- capicxx-someip-runtime.orig/include/CommonAPI/SomeIP/OutputStream.hpp	2025-11-25 12:43:46.180530577 +0200
+++ capicxx-someip-runtime/include/CommonAPI/SomeIP/OutputStream.hpp	2025-11-25 14:13:48.402838947 +0200
@@ -10,6 +10,12 @@
 #ifndef COMMONAPI_SOMEIP_OUTPUTSTREAM_HPP_
 #define COMMONAPI_SOMEIP_OUTPUTSTREAM_HPP_
 
+#if defined(__QNX__)
+    #include <sys/param.h>
+    #include <inttypes.h>
+#endif
+
+
 #include <cstring>
 #include <iomanip>
 #include <limits>
@@ -461,7 +467,7 @@
             _bits = (sizeof(Type_) << 3);
 
         if (currentBit_ == 0 && _bits == (sizeof(Type_) << 3)) {
-        #if __BYTE_ORDER == __LITTLE_ENDIAN
+        #if BYTE_ORDER == LITTLE_ENDIAN
             if (isLittleEndian_) {
                 for (size_t i = 0; i < sizeof(Type_); ++i) {
                     _writeRaw(value.raw_[i]);
@@ -494,7 +500,7 @@
             // Set the source pointer dependend on the byte orders
             byte_t * source(nullptr);
             std::size_t firstUsedByte(((sizeof(Type_) << 3) - _bits) >> 3);
-            #if __BYTE_ORDER == __LITTLE_ENDIAN
+            #if BYTE_ORDER == LITTLE_ENDIAN
             if (isLittleEndian_)
                 source = &value.raw_[firstUsedByte];
             else
@@ -542,7 +548,7 @@
                 readPosition = uint8_t(readPosition + numCopy);
                 if (readPosition == 8) {
                     readPosition = 0;
-                    #if __BYTE_ORDER == __LITTLE_ENDIAN
+                    #if BYTE_ORDER == LITTLE_ENDIAN
                     if (isLittleEndian_)
                         source++;
                     else
@@ -569,7 +575,7 @@
             } value;
             value.typed = _value;
 #if !defined(_WIN32) && !defined(__ANDROID__)
-            if ((__BYTE_ORDER == __LITTLE_ENDIAN) != isLittleEndian_) {
+            if ((BYTE_ORDER == LITTLE_ENDIAN) != isLittleEndian_) {
 #else
             if (!isLittleEndian_) {
 #endif
diff -ruN capicxx-someip-runtime.orig/include/CommonAPI/SomeIP/Watch.hpp capicxx-someip-runtime/include/CommonAPI/SomeIP/Watch.hpp
--- capicxx-someip-runtime.orig/include/CommonAPI/SomeIP/Watch.hpp	2025-11-25 12:43:46.182477309 +0200
+++ capicxx-someip-runtime/include/CommonAPI/SomeIP/Watch.hpp	2025-11-25 18:59:13.284686055 +0200
@@ -3,35 +3,31 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-#if !defined (COMMONAPI_INTERNAL_COMPILATION)
+#if !defined(COMMONAPI_INTERNAL_COMPILATION)
 #error "Only <CommonAPI/CommonAPI.h> can be included directly, this file may disappear or change contents."
 #endif
 
 #ifndef COMMONAPI_SOMEIP_WATCH_HPP_
 #define COMMONAPI_SOMEIP_WATCH_HPP_
 
+#include <CommonAPI/MainLoopContext.hpp>
+#include <CommonAPI/SomeIP/ProxyConnection.hpp>
+#include <CommonAPI/SomeIP/Types.hpp>
 #include <condition_variable>
 #include <functional>
 #include <memory>
 #include <mutex>
 #include <queue>
-
 #include <vsomeip/application.hpp>
 
-#include <CommonAPI/MainLoopContext.hpp>
-#include <CommonAPI/SomeIP/Types.hpp>
-#include <CommonAPI/SomeIP/ProxyConnection.hpp>
-
 namespace CommonAPI {
 namespace SomeIP {
 
 class Connection;
 struct QueueEntry;
 
-class Watch
-        : public CommonAPI::Watch {
-public:
-
+class Watch : public CommonAPI::Watch {
+   public:
     Watch(const std::shared_ptr<Connection>& _connection);
 
     virtual ~Watch();
@@ -44,6 +40,13 @@
     const HANDLE& getAssociatedEvent();
 #endif
 
+#if defined(__QNX__)
+    int readFd_;
+    int writeFd_;
+#else
+    int eventFd_;
+#endif
+
     const std::vector<CommonAPI::DispatchSource*>& getDependentDispatchSources();
 
     void addDependentDispatchSource(CommonAPI::DispatchSource* _dispatchSource);
@@ -60,7 +63,7 @@
 
     void processQueueEntry(std::shared_ptr<QueueEntry> _queueEntry);
 
-private:
+   private:
 #ifdef _WIN32
     int pipeFileDescriptors_[2];
 #else
@@ -77,14 +80,11 @@
 
     std::weak_ptr<Connection> connection_;
 
-#ifdef _WIN32
-    HANDLE wsaEvent_;
-    const int pipeValue_;
-#else
-    const std::uint64_t eventFdValue_;
-
+#if !defined(__QNX__)
+const std::uint64_t eventFdValue_;
 #endif
 
+
     std::mutex lastProcessingMutex_;
     std::chrono::steady_clock::time_point lastProcessing_;
 
@@ -97,7 +97,7 @@
     std::size_t max_queue_size_;
 };
 
-} // namespace IntraP
-} // namespace CommonAPI
+}  // namespace SomeIP
+}  // namespace CommonAPI
 
-#endif // COMMONAPI_SOMEIP_WATCH_HPP_
+#endif  // COMMONAPI_SOMEIP_WATCH_HPP_
