--- capicxx-core-runtime.orig/CMakeLists.txt	2025-11-24 20:44:51.805689698 +0200
+++ capicxx-core-runtime/CMakeLists.txt	2025-11-27 20:04:47.231682561 +0200
@@ -21,6 +21,29 @@
 message(STATUS "This is CMake for Common API C++ Version ${COMPONENT_VERSION}.")
 
 # OS
+# QNX Support
+if(${CMAKE_SYSTEM_NAME} MATCHES "QNX")
+    message(STATUS "Configuring CommonAPI for QNX...")
+
+    add_definitions(-D__QNXNTO__ -D_QNX_SOURCE)
+
+    # Compiler flags for qcc / QCC
+    set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -Vgcc_ntox86_64 -std=c11")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Vgcc_ntox86_64 -std=c++17")
+
+    # # Disable Linux-only calls in the CommonAPI sources
+    # add_definitions(
+    #     -DCAPI_NO_EPOLL
+    #     -DCAPI_NO_EVENTFD
+    #     -DCAPI_NO_TIMERFD
+    # )
+
+    # No libdl on QNX
+    set(DL_LIBRARY "")
+endif()
+
+
+# Linux Support
 set(DL_LIBRARY "")
 if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     set(DL_LIBRARY "dl")
@@ -28,7 +51,11 @@
     # force all use of std::mutx and std::recursive_mutex to use runtime init
     # instead of static initialization so mutexes can be hooked to enable PI as needed
     add_definitions(-D_GTHREAD_USE_MUTEX_INIT_FUNC -D_GTHREAD_USE_RECURSIVE_MUTEX_INIT_FUNC)
-endif (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "QNX")
+    # QNX settings are handled earlier
+endif()
+
 
 ##############################################################################
 
@@ -106,10 +133,15 @@
     add_definitions(-DCOMMONAPI_INTERNAL_COMPILATION -DCOMMONAPI_DLL_COMPILATION)
     add_compile_options(/EHsc /wd4996)
 ELSE ()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -Wall -Wextra -Werror=extra-semi -Wformat -Wformat-security -Wconversion -fexceptions -fstrict-aliasing -fstack-protector-strong -fasynchronous-unwind-tables -fno-omit-frame-pointer -Werror -DCOMMONAPI_INTERNAL_COMPILATION -fvisibility=hidden")
-    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wformat-security -fstack-protector-strong")
+    if(${CMAKE_SYSTEM_NAME} MATCHES "QNX")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -Wall -Wextra -DCOMMONAPI_INTERNAL_COMPILATION -fvisibility=hidden")
+    else()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -Wall -Wextra -Werror=extra-semi -Wformat -Wformat-security -Wconversion -fexceptions -fstrict-aliasing -fstack-protector-strong -fasynchronous-unwind-tables -fno-omit-frame-pointer -Werror -DCOMMONAPI_INTERNAL_COMPILATION -fvisibility=hidden")
+        set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wformat-security -fstack-protector-strong")
+    endif()
 ENDIF(MSVC)
 
+
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DCOMMONAPI_LOGLEVEL=COMMONAPI_LOGLEVEL_${MAX_LOG_LEVEL}")
 
 FIND_PACKAGE(PkgConfig)
